File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/HeroDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class HeroDTO {

    private Long id;
    private String name;
    private int maxHealth;
    private int currentHealth;
    private int attack;
    private int defense;
    private int magic;
    private List<MoveDTO> moves;
}

File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/HeroDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class HeroDto
    {
        [JsonProperty("id")]
        public long id;

        [JsonProperty("name")]
        public string name;

        [JsonProperty("maxHealth")]
        public int maxHealth;

        [JsonProperty("currentHealth")]
        public int currentHealth;

        [JsonProperty("attack")]
        public int attack;

        [JsonProperty("defense")]
        public int defense;

        [JsonProperty("magic")]
        public int magic;

        [JsonProperty("moves")]
        public List<MoveDto> moves = new List<MoveDto>();
    }
}